### PR TITLE
Validations and Fixed Version Issue

### PR DESCRIPTION
**Pull request by Vijay-Giri**

### CODE REVIEW

File: templatestore/views.py [hunk 1]
```diff
@@ -126,13 +176,16 @@ def post_template_view(request):
...
-                minor_version = str(int(minor_version) + 1)
+                ).values("version")
+                version_list = []
+                for version in current_versions:
+                    major, minor = version["version"].split(".")
+                    version_list.append((int(major), int(minor)))
 
-                version = major_version + "." + minor_version
+                version_list.sort(key=lambda ax: (int(ax[0]), int(ax[1])), reverse=True)
+                version = str(version_list[0][0]) + "." + str(version_list[0][1] + 1)
```
Comment (wilspi) [inline templatestore/views.py]: order by id desc -> should also work 
that will give you the recent version

File: templatestore/views.py [hunk 2]
```diff
@@ -299,22 +392,42 @@ def get_template_details_view(request, name, version):
...
+            current_versions = TemplateVersion.objects.filter(template_id=tmp).values(
+                "version"
+            )
+            version_list = []
+            for ver in current_versions:
+                major, minor = ver["version"].split(".")
+                version_list.append((int(major), int(minor)))
+
+            version_list.sort(key=lambda ax: (int(ax[0]), int(ax[1])), reverse=True)
+            new_version = str(float(version_list[0][0] + 1))
```
Comment (wilspi) [inline templatestore/views.py]: same here

File: templatestore/views.py [hunk 3]
```diff
@@ -95,11 +95,53 @@ def post_template_view(request):
...
+            if not len(cfgs):
+                raise (
+                    Exception("Validation: `" + data["type"] + "` is not a valid type")
+                )
+
             sub_types = {cfg.sub_type: cfg for cfg in cfgs}
 
+            invalid_subtypes = set(
+                [s["sub_type"] for s in data["sub_template"]]
+            ).difference(set(sub_types.keys()))
```
Comment (wilspi) [inline templatestore/views.py]: 👍

File: templatestore/views.py (hunk 1 above)
Comment (Vijay-Giri) [reply templatestore/views.py]: yea right ...I will change it

File: templatestore/views.py (hunk 2 above)
Comment (Vijay-Giri) [reply templatestore/views.py]: ohk

**Pull request merged**